src/main/scala/gestalt/package.scala

```diff
@@ -1,38 +1,425 @@
-package scala
+package scala.gestalt
 
-package object gestalt {
-  /** The dummy Toolbox for type checking
-   */
-  implicit val toolbox: Toolbox = null
+import gestalt.core._
 
-  /** The marker for meta block to highlight the different semantics
-   */
-  def meta(body: toolbox.Tree): Nothing = ???
+// package object causes compilation errors
+object api extends Toolbox { pkg =>
+  private val toolbox: ThreadLocal[Toolbox] = new ThreadLocal[Toolbox]
+
+  type Location = core.Location
+
+  def withToolbox[T](tb: Toolbox)(f: => T): T = {
+    toolbox.set(tb)
+    val res = f
+    toolbox.remove()
+
+    res
+  }
+
+  def emptyMods: Mods = toolbox.get.emptyMods.asInstanceOf[Mods]
+
+  // diagnostics
+  def location: Location = toolbox.get.location
+
+  def error(message: String, pos: Pos) = {
+    val tb = toolbox.get
+    tb.error(message, pos.asInstanceOf[tb.Pos])
+  }
+
+  /** stop macro transform */
+  def abort(message: String, pos: Pos): Nothing = {
+    val tb = toolbox.get
+    tb.abort(message, pos.asInstanceOf[tb.Pos])
+  }
+
+  /** generate fresh unique name */
+  def fresh(prefix: String = "$local"): String  = toolbox.get.fresh(prefix)
+
+  /**------------------------------------------------*/
+  // definitions
+  def NewAnonymClass = toolbox.get.NewAnonymClass.asInstanceOf[NewAnonymClassImpl]
```

Comment:
I already reported a bug for Scala here: https://github.com/scala/bug/issues/10318

Dotty supports the syntax without prefix if the extractor is imported.

Repeating the definition of each object is really annoying -- I initially tried that but quickly gave up.